api/v3/PaymentProcessor.php

```diff
@@ -115,3 +115,44 @@ function _civicrm_api3_payment_processor_getlist_defaults(&$request) {
     ),
   );
 }
+
+/**
+ * Action refund.
+ *
+ * @param array $params
+ *
+ * @return array
+ *   API result array.
+ * @throws CiviCRM_API3_Exception
+ */
+function civicrm_api3_payment_processor_refund($params) {
+  $processor = Civi\Payment\System::singleton()->getById($params['payment_processor_id']);
+  $processor->setPaymentProcessor(civicrm_api3('PaymentProcessor', 'getsingle', array('id' => $params['payment_processor_id'])));
+  if (!$processor->supportsRefund()) {
+    throw API_Exception('Payment Processor does not support refund');
+  }
+  $result = $processor->doRefund($params);
+  if (is_a($result, 'CRM_Core_Error')) {
```

Comment:
we don't need this - we don't catch the exception & the api wrapper will - it should never be a CRM_Core_Error

Reply:
Oops will remove it. Actually I took an example of [PaymentProcessor.Pay API](https://github.com/eileenmcnaughton/nz.co.fuzion.omnipaymultiprocessor/blob/master/api/v3/PaymentProcessor/Pay.php#L15) to throw an exception here